app/models/user.rb

```diff
@@ -428,11 +428,19 @@ def disable_2fa!
   # ensures some users talk to a mod before reactivating
   def good_riddance?
     return if is_banned? # https://www.youtube.com/watch?v=UcZzlPGnKdU
+
+    recent_comments_count = comments
+      .where("created_at >= ?", 30.days.ago)
```

Comment:
```suggestion
      .where(created_at: 30.days.ago..)
```

(This might need an extra set of parenthesis around the `30.days.ago`.)

Same edit a couple lines down.